python/ray/autoscaler/_private/commands.py

```diff
@@ -649,7 +649,8 @@ def get_or_create_head_node(config: Dict[str, Any],
                     time.sleep(POLL_INTERVAL)
             cli_logger.newline()
 
-    global_event_system.execute_callback(CreateClusterEvent.head_node_acquired)
+    global_event_system.execute_callback(CreateClusterEvent.head_node_acquired,
+                                         {"node_context": NodeContext(head_node, True)})
```

Comment:
My 2 cents, leaving it on by default even if many customers has not setup sns topics might adds a bit of overhead. Something to consider.

Reply:
Clients who add a custom callback might not necessarily need this detail, but I think the current overhead here is pretty minimal since `NodeContext` is just a wrapper around two values: the node ID (string) and a boolean flag that indicates whether or not this node is a head node.